cirq/protocols/qasm.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import string
+from typing import TYPE_CHECKING, Union, Any, Tuple, TypeVar, Optional, Dict, \
+    Iterable
+
+from typing_extensions import Protocol
+
+from cirq.type_workarounds import NotImplementedType
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    import cirq
+
+TDefault = TypeVar('TDefault')
+
+RaiseTypeErrorIfNotProvided = ([],)  # type: Any
+
+
+class QasmArgs(string.Formatter):
+    """
+    Attributes:
+        precision: The number of digits after the decimal to show for numbers in
+            the text diagram.
```

Comment:
reference to text diagram is slightly odd, maybe "qasm output"?

Reply:
Done.